File: 1251 - 단어 나누기/1251.py
inpt = input()
rslt = "{"

for i in range(len(inpt) - 2):
    for j in range(i + 1, len(inpt) - 1):
        tmp = ""
        for k in range(i, -1, -1):
            tmp += inpt[k]
        for k in range(j, i, -1):
            tmp += inpt[k]
        for k in range(len(inpt) - 1, j, -1):
            tmp += inpt[k]
        
        if rslt > tmp:
            rslt = tmp

print(rslt)

File: 1251-단어_나누기/1251.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    string inpt;
    cin >> inpt;
    string rslt = "{";   // 아스키코드상 알파벳 소문자 바로 뒤

    for (int i = 0; i < inpt.size() - 2; i++){
        for (int j = i + 1; j < inpt.size() - 1; j++){
            string tmp;

            for (int k = i; k >= 0; k--)
                tmp += inpt[k];
            for (int k = j; k > i; k--)
                tmp += inpt[k];
            for (int k = inpt.size() - 1; k > j; k--)
                tmp += inpt[k];
            
            if (rslt > tmp)
                rslt = tmp;
        }
    }

    cout << rslt << '\n';
    return 0;
}
